pubsub/http.go

```diff
@@ -0,0 +1,111 @@
+package pubsub
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/anycable/anycable-go/node"
+	"github.com/anycable/anycable-go/server"
+	"github.com/apex/log"
+)
+
+// HTTPConfig contains HTTP pubsub adapter configuration
+type HTTPConfig struct {
+	// Port to listen on
+	Port int
+	// Path for HTTP broadast
+	Path string
+	// Secret token to authorize requests
+	Secret string
+}
+
+// NewHTTPConfig builds a new config for HTTP pub/sub
+func NewHTTPConfig() HTTPConfig {
+	return HTTPConfig{}
+}
+
+// HTTPSubscriber represents HTTP pub/sub
+type HTTPSubscriber struct {
+	port       int
+	path       string
+	authHeader string
+	server     *server.HTTPServer
+	node       node.AppNode
+	log        *log.Entry
+}
+
+// NewHTTPSubscriber builds a new HTTPSubscriber struct
+func NewHTTPSubscriber(node node.AppNode, config *HTTPConfig) *HTTPSubscriber {
+	authHeader := ""
+
+	if config.Secret != "" {
+		authHeader = fmt.Sprintf("Bearer %s", config.Secret)
+	}
+
+	return &HTTPSubscriber{
+		node:       node,
+		log:        log.WithFields(log.Fields{"context": "pubsub"}),
+		port:       config.Port,
+		path:       config.Path,
+		authHeader: authHeader,
+	}
+}
+
+// Start creates an HTTP server or attaches a handler to the existing one
+func (s *HTTPSubscriber) Start() error {
+	server, err := server.ForPort(strconv.Itoa(s.port))
+
+	if err != nil {
+		return err
+	}
+
+	s.server = server
+	s.server.Title = "Pub/Sub HTTP server"
```

Comment:
Would it be a race condition to mutate title for an existent server? It should handle gracefully a newly created one but what if we attach to the existent?

Reply:
Good point. Probably, we should remove this hack in favor of in-place logging for every server (the only thing we use to for now).